src/api/views/oauthView.py

```diff
@@ -0,0 +1,52 @@
+from api import app 
+from requests_oauthlib import OAuth2Session
+from flask import Flask, request, redirect, session, url_for
+from api.controllers import userController
+
+client_id = ""
+client_secret = ""
+authorization_base_url = 'https://github.com/login/oauth/authorize'
+token_url = 'https://github.com/login/oauth/access_token'
+
+
+@app.route("/auth")
+def auth():
+    github = OAuth2Session(client_id)
+    authorization_url, state = github.authorization_url(authorization_base_url)
+
+    # State is used to prevent CSRF, keep this for later.
+    session['oauth_state'] = state
+    return redirect(authorization_url)
+
+
+@app.route("/callback", methods=["GET"])
+def callback():
+    github = OAuth2Session(client_id, state=session['oauth_state'])
+    token = github.fetch_token(token_url, client_secret=client_secret, authorization_response=request.url)
+
+    # At this point you can fetch protected resources but lets save
+    # the token and show how this is done from a persisted token
+    # in /profile.
+    session['oauth_token'] = token
+
+    return redirect(url_for('.profile'))
+
+@app.route("/profile", methods=["GET"])
+def profile():
+    github = OAuth2Session(client_id, token=session['oauth_token'])
+    user = github.get('https://api.github.com/user').json()
+    username = user['login']
+    email = user['email']
+
+    if(userController.get_user(username=username) = None):
```

Comment:
if(userController.get_user(username=username) == None):

Reply:
Thanks!

Reply:
seems like I messed up.
You didnt need to add request-oauthlib.
I misssed the commit where you already added the relevant library to pipenv.
Please revert that change. after that we have to wait for the api keys to test it.

Reply:
can you also add the code to load the client_id and client secret from the .env file and give an error message if they have not been supplied
For reference look at src/api/models/database.py how the db_url is set and checked

Reply:
I fixed the stuffs